scripts/globals/mobskills.lua

```diff
@@ -375,6 +383,15 @@ xi.mobskills.mobAddBonuses = function(caster, target, dmg, ele) -- used for SMN
     dmg             = math.floor(dmg * dayWeatherBonus)
 
     local burst = calculateMobMagicBurst(caster, ele, target)
+    if
+        burst > 1.0 and
+        caster:getPetID() > 0 -- all pets except charmed pets can get magic burst message, but only with petskill action
+    then
+        target:setLocalVar('abilityMagicBurst', 1)
```

Comment:
Not a fan of using local vars unless absolutely necessary. the two kinds of `skill` we feed in could have something like... `isMagicBurst` and `setMagicBurst`, what do you think?

Reply:
This function does not pass in the skill. If we pass the skill we could set the message directly.

I assumed deciding the message in a single place (final adjustments function) was better

Reply:
I agree that setting the message in final adjustments makes sense, but I think passing in `skill` here and setting the flag to be read later would be ideal

Reply:
I have an idea of what to do to keep the simplicity this PR currently has. Will work on it later today

Reply:
Updated all petskill lua to pass the respective skill object and the magic burst message is set if the skill is there

then in the "final adjustment" functions, I check the message only if it's a pet_finish action.

Spot checked that it still works as expected

Reply:
Touching all these files made me notice that Ifrit's abilities are getting bonus dmg from skillchains:

![image](https://github.com/LandSandBoat/server/assets/131182600/efc613f7-856b-4a6d-a35c-115860f81910)

I assume this isn't correct

Reply:
I don't recall this being a thing, yeah

Reply:
I think we can leave that alone for now and fix it later unless it's egregious levels of extra damage?